spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpointWebIntegrationTests.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.context.properties;
+
+import org.junit.jupiter.api.BeforeEach;
+
+import org.springframework.boot.actuate.endpoint.web.test.WebEndpointTest;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Integration tests for {@link ConfigurationPropertiesReportEndpoint} exposed by Jersey,
+ * Spring MVC, and WebFlux.
+ *
+ * @author Chris Bono
+ */
+class ConfigurationPropertiesReportEndpointWebIntegrationTests {
+
+	private WebTestClient client;
+
+	@BeforeEach
+	void prepareEnvironment(ConfigurableApplicationContext context, WebTestClient client) {
+		TestPropertyValues.of("com.foo.name=fooz", "com.bar.name=barz").applyTo(context);
+		this.client = client;
+	}
+
+	@WebEndpointTest
+	void noFilters() {
+		this.client.get().uri("/actuator/configprops").exchange().expectStatus().isOk().expectBody()
+				.jsonPath("$..beans[*]").value(hasSize(greaterThanOrEqualTo(2))).jsonPath("$..beans['fooDotCom']")
+				.exists().jsonPath("$..beans['barDotCom']").exists();
+	}
+
+	@WebEndpointTest
+	void filterByExactPrefix() {
+		this.client.get().uri("/actuator/configprops/com.foo").exchange().expectStatus().isOk().expectBody()
+				.jsonPath("$..beans[*]").value(hasSize(1)).jsonPath("$..beans['fooDotCom']").exists();
+	}
+
+	@WebEndpointTest
+	void filterByGeneralPrefix() {
+		this.client.get().uri("/actuator/configprops/com.").exchange().expectStatus().isOk().expectBody()
+				.jsonPath("$..beans[*]").value(hasSize(2)).jsonPath("$..beans['fooDotCom']").exists()
+				.jsonPath("$..beans['barDotCom']").exists();
+	}
+
+    @WebEndpointTest
+    void filterByNonExistentPrefix() {
```

Comment:
Should we return 404 in case of no prefix match? If so I need to create a web endpoint extension. Just let me know what the preference is here. Because the prefix can match multiple entries it feels more of a "List findByPrefix" than a "Object findByPrimaryKey" (if you will).

Reply:
A 404 feels right to me. That's what we do in the `env` endpoint for a property that doesn't exist. It too can match multiple entries as the same property may be defined in multiple property sources.